core/native/src/main/scala/zio/internal/PlatformSpecific.scala

```diff
@@ -27,6 +27,12 @@ import zio.internal.tracing.TracingConfig
 
 private[internal] trait PlatformSpecific {
 
+  /**
+   * Adds a shutdown hook that executes the specified action on shutdown.
+   */
+  def addShutdownHook(action: () => Unit): Unit =
```

Comment:
With this added, is it possible to go down to a single `zio.App`, rather than separate versions for JS / JVM?

Reply:
I think so. We may need a separate method with platform specific exit logic like we had in the bootstrap version of the ZIO Test internal test suite. I will work on this in a subsequent PR if that is alright.